crates/proto/src/rr/domain/name.rs

```diff
@@ -1161,15 +1160,15 @@ fn read_inner(
                     Some(0) | None => LabelParseState::Root,
                     Some(byte) if byte & 0b1100_0000 == 0b1100_0000 => LabelParseState::Pointer,
                     Some(byte) if byte & 0b1100_0000 == 0b0000_0000 => LabelParseState::Label,
-                    Some(byte) => return Err(ProtoErrorKind::UnrecognizedLabelCode(byte).into()),
+                    Some(byte) => return Err(DecodeError::UnrecognizedLabelCode(byte)),
                 }
             }
             // labels must have a maximum length of 63
             LabelParseState::Label => {
                 let label = decoder
-                    .read_character_data_max(Some(63))?
+                    .read_character_data()?
```

Comment:
Can we consider this one for a minute? Is this significant to the speed ups? Could we possibly make it more efficient with the bounded length? I like ensuring we don't consume too much into a buffer (i.e. we want to be defensive against malicious actors).

The `verify` does ensure it's also not too large, I just want to ensure we don't accidentally open the door to over-consumption of memory.

Reply:
This is an ingredient to the perf improvements. It is small, like all of the tweaks in this PR, but measurable and non-invasive.

I like your resiliency-oriented approach to reviewing this code; I think there isn't an issue here but I don't want to give the impression that I'm not aligned with your concern.

`read_character_data` is constant-time, because it doesn't actually relocate any of the bytes, it just returns a slice so that the caller can access them. The check that remains still guards against relocating an invalid number of bytes, though since the length is derived from a `u8`, even without this check the maximum number of bytes we could relocate is 255. And even then, I'm not convinced it's even possible to get a length greater than 63 here; we only get to this match arm if the length we're about to read has the top 2 bits unset.

Reply:
That's a great point. Thanks for that explanation.